### DISCUSSION TITLE
Example for Custom Modes / Worker

### DISCUSSION THREAD

**Discussion by klassm** [securingsincity/react-ace#689] [Q&A]
Hi,

I am trying to add a custom mode in my app using react-ace. Now I found lots of issues like #557 (which does not work anymore as there is no longer a ref property), #126 (which does not work because you have to create refs first + there's no ref property). 

Could you somehow give me a hint on how to implement this using the current version?

In addition - how are workers supposed to work for custom languages? I mean brace has all this stuff stringified using some arbitrary script. How is this supposed to work for custom workers?

Thanks for this really cool project :-)
Matthias

**Comment by ColinChartier** [comment]
This is how my AceEditor wrapper looks, in case it is helpful to you:

```
export default class extends React.Component {
    constructor(props) {
        super(props);
        this.aceEditor = React.createRef();
    }

    componentDidMount() {
        this.aceEditor.current.editor.getSession().setMode(new LayerfileMode());
    }

    render() {
        return (
            <div>
                <div>{this.props.chosenRepo}</div>
                <AceEditor
                    theme="github"
                    mode="text"
                    onChange={(val) => console.log(val)}
                    name="UNIQUE_ID_OF_DIV"
                    editorProps={{$blockScrolling: true}}
                    ref={this.aceEditor}
                />,
            </div>
        );
    }
}
```
